packages/cli/src/search/matchFilter.js

```diff
@@ -0,0 +1,65 @@
+// @ts-check
+
+const { formatValue, formatHttpServerRequest } = require('../utils');
+
+/** @typedef {import('./types').Event} Event */
+/** @typedef {function(string,Event[],string):boolean} Matcher */
+
+/** @type {Matcher} */
+const returnValue = (expectedValue, stack) => {
+  const event = stack[stack.length - 1];
+  return formatValue(event.returnValue) === expectedValue;
+};
+
+/** @type {Matcher} */
+const httpServerRequest = (expectedValue, stack) =>
+  !!stack
+    .filter((e) => e.httpServerRequest)
+    .find((e) => formatHttpServerRequest(e) === expectedValue);
```

Comment:
This is a good case for [`some`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some).
```js
const httpServerRequest = (expectedValue, stack) =>
  stack
    .filter((e) => e.httpServerRequest)
    .some((e) => formatHttpServerRequest(e) === expectedValue);
```